algoliasearch-common/src/main/java/com/algolia/search/inputs/query_rules/ConsequenceParams.java

```diff
@@ -129,12 +140,12 @@ public Long getMaxFacetHits() {
     return maxFacetHits;
   }
 
-  public ConsequenceParams setMaxFacetHits(Integer maxFacetHits) {
+  @JsonSetter
+  public ConsequenceParams setMaxFacetHits(Long maxFacetHits) {
     return (ConsequenceParams) super.setMaxFacetHits(maxFacetHits);
   }
 
-  @JsonSetter
-  public ConsequenceParams setMaxFacetHits(Long maxFacetHits) {
+  public ConsequenceParams setMaxFacetHits(Integer maxFacetHits) {
```

Comment:
why not add a deprecation warning on those setters with an `Integer` ?

Reply:
Right, let's deprecate the useless one indeed.

Reply:
Yes! Will do.